hw/hal/src/hal_flash.c

```diff
@@ -338,3 +352,23 @@ hal_flash_ioctl(uint8_t id, uint32_t cmd, void *args)
 {
     return 0;
 }
+
+int
+hal_flash_write_protect(uint8_t id, uint8_t protect)
+{
+    if (NULL == hal_bsp_flash_dev(id)) {
+        return SYS_EINVAL;
+    }
+    if (id / 8 >= sizeof(protected_flash) / sizeof(protected_flash[0])) {
+        assert(0);
```

Comment:
I don't think this function should assert that the ID argument is valid.  I would just let the caller assert success if desired.

Reply:
done